### DISCUSSION TITLE
5 / 10 Minute intervals?

### DISCUSSION THREAD

**Discussion by martin-braun** [theAkito/webmon#46] [Ideas]
It would be nice to heave more interval options. I really want to monitor one endpoint permanently and I wish to do a request every 5 minutes.

**Comment by theAkito** [comment]
Greetings 👋🏻

Thank you for your request.

Did you already check out the Custom Monitoring feature?

It lets you monitor an endpoint continuously every second or every minute.

Would this already solve your issue?

https://webmon.dev.akito.ooo/

When you click through the picture slideshow, there is a screenshot displaying the Custom Monitoring option.

**Reply by martin-braun** [reply]
I've seen that. It felt like an temporary option. If this is persistent, it should be alright indeed. Is it persistent between reboots and such? In my tests, closing the app will stop custom monitoring, I need something 24/7.

**Reply by theAkito** [reply]
Ah, I get it now. Let me think about that.

**Reply by martin-braun** [reply]
@theAkito Thanks, one more question though. Could you consider this app to have a configurable alarm? I literally want my phone to ring at me if a service is down, so that I don't miss its outage.

**Reply by theAkito** [reply]
Good idea. I would need to look up how it works with the Android alarm system.

The problem with punctuality & reliability in Android is though, that some phones are very restrictive on keeping something running in background, so I hope, using the alarm system would work as expected.

https://dontkillmyapp.com/

Most of the time, when people report, that some alarm or notification did not appear in time, it's not due to the app, but due to the Android derivative on the phone.

**Reply by martin-braun** [reply]
@theAkito I turn off battery optimization for webmon in that case. Afaik you are using the background service that is very restrictive, also with the 15 minute cycle. But giving the option to use a foreground service instead (with persistent notification) would be the solution to the problem to go down to 5 minutes and offer alarm sounds without any delay. However, the app needs to be very performance efficient to not make an impact.

Let me tell you, for me, this wouldn't be a deal breaker. Right now I use cron-job.org which can only send emails, but my email client drains good 5% through a battery cycle. I built a proxy that uses the cron-job API and returns HTTP status code 200 only when all my cron-jobs run well. So if you manage to get webmon with one endpoint to only 2% drainage in a day with 5 minute intervals, that would be fulfilling for me. 

I like it simple. I don't need to consume my emails on my phone, and if I really want to I can still use a web client for that.

**Reply by theAkito** [reply]
The option to use the foreground service is already implemented. Not sure how performant all that is, though.

Also not sure how to measure such performance with tons of different smartphones with different specs, quirks & problems on the market... 🤔

**Comment by theAkito** [comment]
> It would be nice to heave more interval options. I really want to monitor one endpoint permanently and I wish to do a request every 5 minutes.

Well, crap.

https://github.com/theAkito/webmon/blob/928390ee30af4c6d4dda6aa4ef0353da9bc04340/app/src/main/kotlin/ooo/akito/webmon/utils/BackgroundCheckInterval.kt#L5-L6

Obviously, totally forgot about that one.

https://stackoverflow.com/a/55349427

This also answers the question about performance.

Some other answers propose some "hacks", but I think they are too much & it probably has a reason the minimum is 15 minutes for performance reasons.

As pointed out earlier, this app already is heavy on resources from the perspective of a lot of phones, which many people do not agree with, so I don't want to make it even heavier & more prone to being shut off, which would reduce reliability & this app must be reliable to be worth a damn.

It also does not seem to be worth the effort right now, because I would need to make a big deal, just to have it every 5 or 10 minutes, rather than every 15 minutes.

**Reply by martin-braun** [reply]
First of all, thank you for looking into this. I respect your decision and it might not be worth the effort, in a perfect world it would still be possible to have an alternative route that hacks its way out of this limitation. It's certainly not heavier than running a full-fledged email client, so I guess it really depends on the use-case and situation. Maybe getting an alert 10 minutes later should not be big of a deal. If it is, I would probably consider real-time monitoring for enterprise systems. I still hope you can understand my frustration with this, but yea, some hacks look promising, but there is no guarantee they will keep working.

**Reply by theAkito** [reply]
Maybe I will look into some hacks & reconsider, how big of a deal it is. I am in reality neither for or against this feature & it's sometimes tough to balance the pros & cons against each other. 🙂

I just definitely don't want to do it, if the hacks are really weird & "big", just to have an interval even more frequent.

Do you by any chance know how e-mail clients are doing it? Does K-9 have a solution for this? I also might look into that & see how they are doing. I doubt, they use a while sleep loop, though. 😆
